internal/proxy/impl.go

```diff
@@ -1697,6 +1697,138 @@ func (node *Proxy) ShowPartitions(ctx context.Context, request *milvuspb.ShowPar
 	return spt.result, nil
 }
 
+func (node *Proxy) getMsgBase() (*commonpb.MsgBase, error) {
+	msgID, err := node.idAllocator.AllocOne()
+	if err != nil {
+		return nil, err
+	}
+	timestamp, err := node.tsoAllocator.AllocOne()
+	if err != nil {
+		return nil, err
+	}
+	return &commonpb.MsgBase{
+		MsgID:     msgID,
+		Timestamp: timestamp,
+		SourceID:  Params.ProxyCfg.GetNodeID(),
+	}, nil
+}
+
+func (node *Proxy) getCollectionProgress(ctx context.Context, request *milvuspb.GetLoadingProgressRequest, collectionID int64) (int64, error) {
+	resp, err := node.queryCoord.ShowCollections(ctx, &querypb.ShowCollectionsRequest{
+		Base: &commonpb.MsgBase{
+			MsgType:   commonpb.MsgType_ShowCollections,
+			MsgID:     request.Base.MsgID,
+			Timestamp: request.Base.Timestamp,
+			SourceID:  request.Base.SourceID,
+		},
+		CollectionIDs: []int64{collectionID},
+	})
+	if err != nil {
+		return 0, err
+	}
+	if len(resp.InMemoryPercentages) == 0 {
+		return 0, errors.New("fail to show collections from the querycoord, no data")
+	}
+	return resp.InMemoryPercentages[0], nil
+}
+
+func (node *Proxy) getPartitionProgress(ctx context.Context, request *milvuspb.GetLoadingProgressRequest, collectionID int64) (int64, error) {
+	IDs2Names := make(map[int64]string)
+	partitionIDs := make([]int64, 0)
+	for _, partitionName := range request.PartitionNames {
+		partitionID, err := globalMetaCache.GetPartitionID(ctx, request.CollectionName, partitionName)
+		if err != nil {
+			return 0, err
+		}
+		IDs2Names[partitionID] = partitionName
+		partitionIDs = append(partitionIDs, partitionID)
+	}
+	resp, err := node.queryCoord.ShowPartitions(ctx, &querypb.ShowPartitionsRequest{
+		Base: &commonpb.MsgBase{
+			MsgType:   commonpb.MsgType_ShowPartitions,
+			MsgID:     request.Base.MsgID,
+			Timestamp: request.Base.Timestamp,
+			SourceID:  request.Base.SourceID,
+		},
+		CollectionID: collectionID,
+		PartitionIDs: partitionIDs,
+	})
+	if err != nil {
+		return 0, err
+	}
+	if len(resp.InMemoryPercentages) != len(partitionIDs) {
+		return 0, errors.New("fail to show partitions from the querycoord, invalid data num")
+	}
+	var progress int64
+	for _, p := range resp.InMemoryPercentages {
+		progress += p
+	}
+	progress /= int64(len(partitionIDs))
+	return progress, nil
+}
+
+func (node *Proxy) GetLoadingProgress(ctx context.Context, request *milvuspb.GetLoadingProgressRequest) (*milvuspb.GetLoadingProgressResponse, error) {
+	if !node.checkHealthy() {
+		return &milvuspb.GetLoadingProgressResponse{Status: unhealthyStatus()}, nil
+	}
+	method := "GetLoadingProgress"
+	tr := timerecord.NewTimeRecorder(method)
+	log.Debug(method, zap.String("collection_name", request.CollectionName),
```

Comment:
use at least info log in all DDL path. and the log format should be something like:
	log.Info(
		rpcReceived(method),
		zap.String("traceID", traceID),
		zap.String("role", typeutil.ProxyRole),
		zap.Any("request", request))